lib/vrl/stdlib/src/parse_json.rs

```diff
@@ -7,6 +10,63 @@ fn parse_json(value: Value) -> Resolved {
     Ok(value)
 }
 
+fn parse_json_with_depth(value: Value, max_depth: Value) -> Resolved {
+    let bytes = value.try_bytes()?;
+    let parsed_depth = positive_int_depth(max_depth)?;
+
+    let raw_value: Box<RawValue> =
+        serde_json::from_slice::<_>(&bytes).map_err(|e| format!("unable to read json: {}", e))?;
+
+    let res = parse_once_with_depth(raw_value, parsed_depth)
+        .map_err(|e| format!("unable to parse json with max depth: {}", e))?;
+    Ok(Value::from(res))
+}
+
+fn positive_int_depth(value: Value) -> std::result::Result<i64, ExpressionError> {
+    let res = value.try_integer()?;
+    if !(1..=128).contains(&res) {
+        Err(ExpressionError::from(format!(
+            "max_depth value should be greater than 0 and less than 128, got {}",
+            res
+        )))
+    } else {
+        Ok(res)
+    }
+}
+
+fn parse_once_with_depth(
+    value: Box<RawValue>,
+    max_depth: i64,
+) -> std::result::Result<JsonValue, Error> {
+    if value.get().starts_with('{') {
+        if max_depth == 0 {
+            serde_json::value::to_value(value.to_string())
```

Comment:
Does this actually work? This leaves me all kinds of confused what is going on here, so maybe some code comments would help explain the logic.

Reply:
Sure, no problem. I will try to clarify the logic.